modules/tftp/server.rb

```diff
@@ -150,6 +150,21 @@ def pxeconfig_file(mac)
     end
   end
 
+  def self.fetch_boot_image(dst, src)
+
+    # Verify dst is a valid directory
+    dst_file = Pathname.new(dst).cleanpath
+    dst_path = Pathname.new(dst.delete_suffix(".iso"))
+
+    FileUtils.mkdir_p dst_path.parent
+    choose_protocol_and_fetch(src, dst_file).join
+    # extract iso
+    extract_task = ::Proxy::ArchiveExtract.new(dst_file, dst_path).start
```

Comment:
This IMHO introduces command execution flaw, Pathname and cleanpath only cleans pathnames, but any arbitrary path can be passed:

```
>> Pathname.new("/anything/can/be/passed").cleanpath
=> #<Pathname:/anything/can/be/passed>
```

I do not have any attack vector in my hands now, but clearly passing `/etc/passwd` into any extracting utility is not what we want to allow. I think the path must be enforced to be relative, then cleaned (stripped of any `..`) and then appended to some kind of base dir constant.

Reply:
I see what you mean.

Maybe it might be better to set the base-path (here: `/var/www/html/pub/installation_media`) in the foreman-proxy configuration rather than by the request.
This has the downside that we cannot customize it at runtime, but I am not sure if this really is necessary.

Reply:
I've implemented a constant base-path to minimise the amount of parameters for this feature. 
Moreover, a constantly defined path makes it easier to know the exact path on the foreman-side. This important to render the templates correctly.